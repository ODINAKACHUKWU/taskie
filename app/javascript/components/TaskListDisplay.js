import React, { Fragment } from "react";
import PropTypes from "prop-types";
import Task from "./Task";

const defaultProps = {
  taskList: [],
};

function TaskListDisplay({ taskList = defaultProps.taskList }) {
  return (
    <Fragment>
      {taskList?.map((task) => (
        <Task key={task.slug} task={task} />
      ))}
    </Fragment>
  );
}

TaskListDisplay.propTypes = {
  taskList: PropTypes.array.isRequired,
};

export default TaskListDisplay;
