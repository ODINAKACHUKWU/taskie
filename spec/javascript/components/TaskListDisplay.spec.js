import "jsdom-global/register";
import React from "react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import TaskListDisplay from "../../../app/javascript/components/TaskListDisplay";

const tasks = [
  {
    title: "Task 1",
    description: "This is task 1",
    avatar_url: "",
    slug: "task-1-1",
  },
  {
    title: "Task 2",
    description: "This is task 2",
    avatar_url: "",
    slug: "task-2-2",
  },
];

describe("rendering TaskListDisplay component", () => {
  it("renders TaskListDisplay component without crashing", () => {
    shallow(<TaskListDisplay taskList={tasks} />);
  });

  it("renders all passed Task components without crashing", () => {
    const wrapper = mount(<TaskListDisplay taskList={tasks} />);
    const cards = wrapper.find(".task-card");

    expect(cards.length).toEqual(tasks.length);
  });
});

describe("passing props", () => {
  it("accepts taskList prop", () => {
    const wrapper = mount(<TaskListDisplay taskList={tasks} />);

    expect(wrapper.props().taskList).toEqual(tasks);
  });
});

describe("snapshot", () => {
  it("TaskListDisplay snapshot", () => {
    const tree = shallow(<TaskListDisplay taskList={tasks} />);

    expect(toJSON(tree)).toMatchSnapshot();
  });
});
