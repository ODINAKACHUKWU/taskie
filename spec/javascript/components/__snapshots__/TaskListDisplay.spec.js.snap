// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot TaskListDisplay snapshot 1`] = `
<Fragment>
  <Task
    key="task-1-1"
    task={
      Object {
        "avatar_url": "",
        "description": "This is task 1",
        "slug": "task-1-1",
        "title": "Task 1",
      }
    }
  />
  <Task
    key="task-2-2"
    task={
      Object {
        "avatar_url": "",
        "description": "This is task 2",
        "slug": "task-2-2",
        "title": "Task 2",
      }
    }
  />
</Fragment>
`;
